dashboard/app/models/user.rb

```diff
@@ -835,8 +835,8 @@ def User.track_proficiency(user_id, script_id, level_id)
 
   # returns whether a new level has been completed and asynchronously enqueues an operation
   # to update the level progress.
-  def track_level_progress_async(script_level:, new_result:, submitted:, level_source_id:)
-    level_id = script_level.level_id
+  def track_level_progress_async(script_level:, new_result:, submitted:, level_source_id:, level:)
+    level_id = level.try(:id) || script_level.level_id
```

Comment:
@balderdash does your merging of completion status also work with the submitted field for submittable levels?